### PR TITLE
#166789999 Users Should Be Able To Rate An Article

### PR DESCRIPTION
**Pull request by niyoceles**
#### What does this PR do?
- user able to rate an article
#### Description of Task to be completed
- Create a relationship database
- Write a test
- Write a documentation
#### How should this be manually tested?
1. Run the following commands in Terminal
```
git clone git@github.com:andela/ah-92explorers-backend.git
cd ah-92explorers-backend
git checkout ft-user-rating-article-166789999
```
2. npm install -g sequelize-cli

3. run `npm install`

4. Install your local database in Postgres

4. Rename `.env.sample` to `.env` Open it and replace  `database-name` your created database

5. run in terminal `sequelize db:migrate

6. run `npm run start`

7. Using Postman
- `POST` `localhost:3000/api/articles/<slug>/rate`

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#166789999](https://www.pivotaltracker.com/story/show/166789999)
#### Screenshots (if appropriate)
`validate a rate`
Screen Shot 2019-07-22 at 09 49 03

`validate slug`
Screen Shot 2019-07-22 at 09 51 47

`rate successful`
Screen Shot 2019-07-22 at 16 50 50

#### Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

### CODE REVIEW

**Review by kodek-sleuth — DISMISSED** [review]

**Review by Hervera — DISMISSED** [review]
Ready to be merged

**Review by IsaiahRn — DISMISSED** [review]

File: controllers/rating.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ */
+class Rating {
+  /**
+   * @param {object} req Request sent to the route
+   * @param {object} res Response from server
+   * @returns {object} Object representing the response returned
+   */
+  static async rateArticle(req, res) {
+    try {
+      const findArticle = await articles.findOne({
```
Comment (niyoceles) [inline controllers/rating.js]: rename findArticle to `article`

File: controllers/rating.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        where: { slug: req.params.slug }, attributes: ['id', 'title', 'slug']
+      });
+      if (!findArticle) {
+        res.status(400).json({ error: 'Slug is not found' });
+      }
+      const findAuthor = await users.findOne({
```
Comment (niyoceles) [inline controllers/rating.js]: rename findAuthor  to `author`

File: controllers/rating.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        where: { email: req.decoded.email }, attributes: ['id', 'firstName', 'lastName']
+      });
+
+      const findRate = await rating.findOne({
```
Comment (niyoceles) [inline controllers/rating.js]: Remove checking existing rating

File: controllers/rating.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            message: 'rate article success',
+            rating: {
+              rating: createRate.rating,
+              author: `${firstName} ${lastName}`,
+              title: findArticle.title,
+            }
+          });
+        }
+      }
+      return res.status(200).json({
```
Comment (niyoceles) [inline controllers/rating.js]: Not necessary to return the  existing rate

**Review by IsaiahRn — DISMISSED** [review]
It can be merged

**Review by Hervera — APPROVED** [review]
Ready to be merged

**Review by elemanhillary-zz — DISMISSED** [review]
ready for merge

**Review by Hervera — CHANGES_REQUESTED** [review]
Rework on making average of ratings

**Review by Hervera — DISMISSED** [review]
Ready to be approved

**Review by IsaiahRn — DISMISSED** [review]

**Review by elemanhillary-zz — DISMISSED** [review]
ready for merge

**Review by elemanhillary-zz — APPROVED** [review]
ready for merge

**Review by IsaiahRn — APPROVED** [review]

**Review by Hervera — APPROVED** [review]
Ready to be approved

**Pull request merged**